### PR TITLE
feat: adds Kuma GUI update PR job (TESTING GITHUB ACTIONS WORKFLOW)

### PR DESCRIPTION
**Pull request by kleinfreund**
# DO NOT MERGE THIS PR

I’m testing a GitHub Actions workflow and have pushed a branch to kuma-gui’s original repository and not a fork. This is done on purpose so I can test the workflow (at least I hope that’s how it works.

---

Adds a new GitHub Actions workflow which automatically creates a pull request in Kuma for updating Kuma GUI.

Limits the main workflow’s triggers for pushes to the project’s default branch.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: .github/workflows/create-gui-pr.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate GitHub App token
+        # https://github.com/tibdex/github-app-token
+        # Note: Specifically references https://github.com/tibdex/github-app-token/commit/f717b5ecd4534d3c4df4ce9b5c1c2214f0f7cd06 (v1.6.0) to ensure that this exact version of the workflow action is used. This is a security precaution as automatically opting into newer versions of the action could leak or get the app token stolen.
+        uses: tibdex/github-app-token@f717b5ecd4534d3c4df4ce9b5c1c2214f0f7cd06
+        id: generate-token
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
```
Comment (michaelbeaumont) [inline .github/workflows/create-gui-pr.yml:33]: I suspect you'll need to [set `repository`](https://github.com/tibdex/github-app-token#example-workflow) to get a token with permissions for another repo

Comment (kleinfreund) [reply .github/workflows/create-gui-pr.yml:33]: That sounds reasonable, but it does seem to work already: https://github.com/kumahq/kuma/pull/4925

Comment (michaelbeaumont) [reply .github/workflows/create-gui-pr.yml:33]: Yeah it does

**Comment by kleinfreund** [comment]
Closing in favour of #370.

**Pull request closed without merging**